Source/JavaScriptCore/runtime/WeakMapPrototype.cpp

```diff
@@ -76,7 +76,9 @@ JSC_DEFINE_HOST_FUNCTION(protoFuncWeakMapDelete, (JSGlobalObject* globalObject,
     if (!map)
         return JSValue::encode(jsUndefined());
     JSValue key = callFrame->argument(0);
-    return JSValue::encode(jsBoolean(key.isObject() && map->remove(asObject(key))));
+    if (UNLIKELY(!canBeHeldWeakly(key)))
+        return JSValue::encode(jsBoolean(false));
```

Comment:
Is it worth doing the extra uid load and isDerred load? Maybe it's faster to just do the hash table lookup since it's unlikely that anyone is a registered symbol. And we're guaranteed to not have the key in the set if it's registered.

Reply:
Maybe we want some function for this that's different than canBeHeldWeakly since we use that necessarily along the add path

Reply:
Probably, we should just check it is a cell. That's enough.